edward/inferences/inference.py

```diff
@@ -258,3 +284,48 @@ def finalize(self):
     """
     if self.logging:
       self.train_writer.close()
+
+  def set_log_variables(self, logvars=None, log_max_scalers_per_var=None):
+    """Logs variables to TensorBoard
+
+     For each variable in logvars, creates 'scalar' and / or 'histogram' by
+     calling `tf.summary.scalar` or `tf.summary.histogram`
+
+     if logvars is None, automatically log all latent variables that have been
+     given non-default names.  If logvars is [], no logging will be created.
+
+     Parameters
+     ----------
+     logvars : list, optional
+       A list of variables to be logged
+     log_max_scalers_per_var : int, default None
+       Enables logging of individual values from 1 dimensional variables up to
+       a maximum dimension, if None will log all dimensions.
+
+     Returns
+     -------
+     None
+
+    """
+    if logvars is None:
+      logvars = []
+      for k in self.latent_vars:
+        logvars += get_variables(self.latent_vars[k])
```

Comment:
variables can also exist as model parameters. I recommend an additional loop applying this to keys of `self.data` and the priors (which are keys of `self.latent_vars`).